python/tk_framework_adobe_utils/plugin_bootstrap.py

```diff
@@ -64,8 +64,19 @@ def bootstrap(root_path, port, engine_name, app_id):
     # get a handle on the newly bootstrapped engine
     engine = sgtk.platform.current_engine()
 
-    from sgtk.platform.qt import QtGui
-    from sgtk.platform.engine_logging import ToolkitEngineHandler
+    try:
+        # First, make sure we can import PySide.
+        # If not, there's no need to continue.
+        from sgtk.platform.qt import QtGui
+
+        if QtGui is None:
+            raise ImportError("PySide not found in Toolkit.")
+    except ImportError:
+        sys.stdout.write("[ERROR]: %s" % (traceback.format_exc(),))
+        sys.stdout.flush()
+        sys.exit(EXIT_STATUS_NO_PYSIDE)
+    finally:
+        from sgtk.platform.engine_logging import ToolkitEngineHandler
```

Comment:
Why have this in a finally block? I think we can have this line unchanged since the except "exits"

```suggestion

    from sgtk.platform.engine_logging import ToolkitEngineHandler
```

Reply:
Coding preferences 🤓